File: BM.DAL/BookCategoryService.cs
using System.Data;
using System.Data.SqlClient;
using BookManagement.Model;
using BookManagement.DAL.Utility;

namespace BookManagement.DAL
{
    public class BookCategoryService
    {
        /// <summary>
        /// 获取所有图书类别
        /// </summary>
        public DataTable GetAllCategories()
        {
            string sql = "SELECT CategoryId, CategoryName, Description FROM BookCategory ORDER BY CategoryId";
            DataSet ds = SQLHelper.ExecuteDataSet(sql);
            return ds != null && ds.Tables.Count > 0 ? ds.Tables[0] : null;
        }

        /// <summary>
        /// 添加图书类别
        /// </summary>
        public int AddCategory(BookCategory category)
        {
            string sql = $"INSERT INTO BookCategory (CategoryName, Description) VALUES ('{category.CategoryName}', '{category.Description}')";
            return SQLHelper.ExecuteNonQuery(sql);
        }

        /// <summary>
        /// 更新图书类别
        /// </summary>
        public int UpdateCategory(BookCategory category)
        {
            string sql = $"UPDATE BookCategory SET CategoryName='{category.CategoryName}', Description='{category.Description}' WHERE CategoryId={category.CategoryId}";
            return SQLHelper.ExecuteNonQuery(sql);
        }

        /// <summary>
        /// 删除图书类别
        /// </summary>
        public int DeleteCategory(int categoryId)
        {
            // 注意：在实际项目中，删除前应检查该类别下是否有图书
            string sql = $"DELETE FROM BookCategory WHERE CategoryId={categoryId}";
            return SQLHelper.ExecuteNonQuery(sql);
        }
    }
}

File: Python/dal/book_category_service.py
from dal.db_helper import DBHelper
from models.book_category import BookCategory


class BookCategoryService:
    def __init__(self):
        self.db = DBHelper()

    def get_all_categories(self) -> list:
        sql = "SELECT CategoryId, CategoryName, Description FROM BookCategory ORDER BY CategoryId"
        rows, _ = self.db.execute_data_set(sql)
        return rows

    def add_category(self, category: BookCategory) -> int:
        sql = "INSERT INTO BookCategory (CategoryName, Description) VALUES (?, ?)"
        return self.db.execute_non_query(sql, (category.category_name, category.description))

    def update_category(self, category: BookCategory) -> int:
        sql = "UPDATE BookCategory SET CategoryName=?, Description=? WHERE CategoryId=?"
        return self.db.execute_non_query(sql, (category.category_name, category.description, category.category_id))

    def delete_category(self, category_id: int) -> int:
        sql = "DELETE FROM BookCategory WHERE CategoryId=?"
        return self.db.execute_non_query(sql, (category_id,))